Teuchos::ParameterList: Shadowing warning

## Bug Report
@trilinos/teuchos @rrdrake 

### Description
```
In file included from .../packages/teuchos/parameterlist/test/ParameterList/ParameterList_XMLDuplicateSublists.cpp:43:0:
.../packages/teuchos/parameterlist/src/Teuchos_ParameterList.hpp: In member function ‘void Teuchos::ParameterList::recursivelySetValidator(const Teuchos::RCP<const Teuchos::ParameterEntryValidator>&, int)’:
...packages/teuchos/parameterlist/src/Teuchos_ParameterList.hpp:1027:23: warning: declaration of ‘entry’ shadows a member of 'this' [-Wshadow]
       ParameterEntry *entry = this->getEntryPtr(entry_name);
                       ^
```

--- a/packages/teuchos/parameterlist/src/Teuchos_ParameterList.hpp
+++ b/packages/teuchos/parameterlist/src/Teuchos_ParameterList.hpp
@@ -1024,9 +1024,9 @@ void ParameterList::recursivelySetValidator(
     if (this->isSublist(entry_name) && depth > 0){
       this->sublist(entry_name).recursivelySetValidator<T>(validator, depth - 1);
     } else{
-      ParameterEntry *entry = this->getEntryPtr(entry_name);
-      if (entry->isType<T>()){
-        entry->setValidator(validator);
+      ParameterEntry *theEntry = this->getEntryPtr(entry_name);
+      if (theEntry->isType<T>()){
+        theEntry->setValidator(validator);
       }
     }
   }

